omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/PatientIdentifierResource1_8.java

```diff
@@ -89,11 +90,12 @@ public DelegatingResourceDescription getCreatableProperties() {
 	public void setIdentifierType(PatientIdentifier instance, PatientIdentifierType identifierType) {
 		String uuid = identifierType.getUuid();
 		String name = identifierType.getName();
-		if (name != null && !name.isEmpty()) {
-			instance.setIdentifierType(Context.getPatientService().getPatientIdentifierTypeByName(name));
-		} else if (uuid != null && !uuid.isEmpty()) {
+		if (!StringUtils.isEmpty(uuid)) {
 			instance.setIdentifierType(Context.getPatientService().getPatientIdentifierTypeByUuid(uuid));
 		}
+		if (instance.getIdentifierType() == null && !StringUtils.isEmpty(name)) {
```

Comment:
The original code would call instance.setIdentifierType() with the identifier type as represented by name. With your changes, as long as instance.getIdentifierType() is not null, instance.setIdentifierType() would not be called to set the identifier type in name.